snapshots/windows/windows.go

```diff
@@ -349,24 +358,29 @@ func (s *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 			return fmt.Errorf("failed to create snapshot: %w", err)
 		}
 
-		if kind != snapshots.KindActive {
-			return nil
-		}
-
-		log.G(ctx).Debug("createSnapshot active")
+		log.G(ctx).Debug("createSnapshot")
 		// Create the new snapshot dir
 		snDir := s.getSnapshotDir(newSnapshot.ID)
 		if err = os.MkdirAll(snDir, 0700); err != nil {
 			return err
 		}
 
-		// IO/disk space optimization
-		//
-		// We only need one sandbox.vhdx for the container. Skip making one for this
-		// snapshot if this isn't the snapshot that just houses the final sandbox.vhd
-		// that will be mounted as the containers scratch. Currently the key for a snapshot
-		// where a layer will be extracted to will have the string `extract-` in it.
-		if !strings.Contains(key, snapshots.UnpackKeyPrefix) {
+		if strings.Contains(key, snapshots.UnpackKeyPrefix) {
+			// IO/disk space optimization: Do nothing
+			//
+			// We only need one sandbox.vhdx for the container. Skip making one for this
+			// snapshot if this isn't the snapshot that just houses the final sandbox.vhd
+			// that will be mounted as the containers scratch. Currently the key for a snapshot
+			// where a layer will be extracted to will have the string `extract-` in it.
+		} else if len(newSnapshot.ParentIDs) == 0 {
+			// A parentless snapshot is just a bind-mount to a directory named
+			// "Files". When committed, there'll be some post-processing to fill in the rest
+			// of the metadata.
+			filesDir := filepath.Join(snDir, "Files")
+			if err := os.MkdirAll(filesDir, 0700); err != nil {
+				return err
+			}
+		} else {
```

Comment:
The first case is kinda wonky now (to me at least), could we re-order this so one of the cases isn't just housing a comment

Reply:
The boolean logic got ugly when I tried that before. We could invert the first test and push everything a level deeper, or copy the inverted test into each following if/elseif.

Probably the least-ugly option would be to push all the logic in the second and third branches out into a new "ensureDataStorage" function (or two functions...). Although that loses a nicety of this PR that the `else` branch is unmodified, which helps keep the diff readable.

As a quick fix, looking at the current logic, that first branch could probably `return nil`, since we're already inside a lambda which is in-effect `storage.CreateSnapshot` followed by `ensureDataStorage`, and this if-else block is followed by `return nil` already.

In fact, if the `if` and `else if` ended with `return nil`, then the `else` block can be _unindented_ a level in-place. That's actually pretty nice, but again costs the PR a modicum of "did this code change?" readability. (Probably a worthwhile sacrifice, and turning on ignore-whitespace would resolve that anyway). As long as we won't ever need to do anything common to these directories after this step, I guess.